framework/http/FileStream.php

```diff
@@ -0,0 +1,263 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\http;
+
+use Psr\Http\Message\StreamInterface;
+use Yii;
+use yii\base\BaseObject;
+use yii\base\ErrorHandler;
+use yii\base\InvalidConfigException;
+
+/**
+ * FileStream represents file stream.
+ *
+ * Example:
+ *
+ * ```php
+ * $stream = new FileSteam([
+ *     'filename' => '@app/files/items.txt',
+ *     'mode' => 'w+',
+ * ]);
+ *
+ * $stream->write('some content...');
+ * $stream->close();
+ * ```
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @since 2.1.0
+ */
+class FileStream extends BaseObject implements StreamInterface
+{
+    /**
+     * @var string file or stream name.
+     * Path alias can be used here, for example: '@app/runtime/items.csv'.
+     * This field can also be PHP stream name, e.g. anything which can be passed to `fopen()`, for example: 'php://input'.
+     */
+    public $filename;
```

Comment:
I would prefer to keep setup via `$config` array.
In this way instance will be configurable via regular application config if necessary.